lib/go/thrift/simple_server.go

```diff
@@ -149,8 +151,10 @@ func (p *TSimpleServer) Serve() error {
 }
 
 func (p *TSimpleServer) Stop() error {
-	p.quit <- struct{}{}
-	p.serverTransport.Interrupt()
+	if atomic.CompareAndSwapInt64(&p.stopped, 0, 1) {
```

Comment:
Okay the example makes it clear.  I thought I found documentation that said it returns the previous value... I guess not!